dspace-api/src/test/java/org/dspace/xmlworkflow/XmlWorkflowServiceIT.java

```diff
@@ -0,0 +1,109 @@
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.builder.ClaimedTaskBuilder;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.EPersonBuilder;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.core.Constants;
+import org.dspace.discovery.IndexingService;
+import org.dspace.eperson.EPerson;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.xmlworkflow.factory.XmlWorkflowServiceFactory;
+import org.dspace.xmlworkflow.service.XmlWorkflowService;
+import org.dspace.xmlworkflow.state.Workflow;
+import org.dspace.xmlworkflow.storedcomponents.ClaimedTask;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * IT for {@link XmlWorkflowServiceImpl}
+ *
+ * @author Maria Verdonck (Atmire) on 14/12/21
+ */
+public class XmlWorkflowServiceIT extends AbstractIntegrationTestWithDatabase {
+
+    protected XmlWorkflowService xmlWorkflowService = XmlWorkflowServiceFactory.getInstance().getXmlWorkflowService();
+    protected IndexingService indexer = DSpaceServicesFactory.getInstance().getServiceManager()
+                                                             .getServiceByName(IndexingService.class.getName(),
+                                                                 IndexingService.class);
+    protected AuthorizeService authorizeService = AuthorizeServiceFactory.getInstance().getAuthorizeService();
+
+    /**
+     * Test to verify that if a user submits an item into the workflow, then it gets rejected that the submitter gets
+     * write access back on the item
+     *
+     * @throws Exception
+     */
+    @Test
+    public void workflowUserRejectsItemTheySubmitted_ItemShouldBeEditable() throws Exception {
+        context.turnOffAuthorisationSystem();
+        EPerson submitter = EPersonBuilder.createEPerson(context).withEmail("submitter@example.org").build();
+        context.setCurrentUser(submitter);
+        Community community = CommunityBuilder.createCommunity(context)
+                                              .withName("Parent Community")
+                                              .build();
+        Collection colWithWorkflow = CollectionBuilder.createCollection(context, community)
+                                                      .withName("Collection WITH workflow")
+                                                      .withWorkflowGroup(1, submitter)
+                                                      .build();
+        Workflow workflow = XmlWorkflowServiceFactory.getInstance().getWorkflowFactory().getWorkflow(colWithWorkflow);
+        ClaimedTask taskToReject = ClaimedTaskBuilder.createClaimedTask(context, colWithWorkflow, submitter)
+                                                     .withTitle("Test workflow item to reject").build();
```

Comment:
@KevinVdV : You look to be leaving the authorization turned **off** for the entirety of the test...that can result in odd/unexpected behaviors.  You should ensure you are turning the auth system back *on* immediately after this line of code, as you only need it turned off to create test content.

Reply:
@tdonohue You are right, I missed that part in my internal review. Quickly fixed it with a commit.